### PR TITLE
Fix `CooMatrix.to_dense` for duplicate indices

### PR DESCRIPTION
**Pull request by hvy**
Fixes https://github.com/chainer/chainer/issues/6543.

### CODE REVIEW

File: chainer/utils/sparse.py [hunk 1]
```diff
@@ -53,22 +63,34 @@ def __init__(self, data, row, col, shape, order=None,
...
-            x[self.row[:nnz], self.col[:nnz]] = data[:nnz]
-            return x
-        if data.ndim == 2:
-            nb = data.shape[0]
-            x = xp.zeros((nb, self.shape[0], self.shape[1]),
-                         dtype=data.dtype)
-            for i in range(nb):
-                nnz = xp.count_nonzero(data[i])
-                x[i, self.row[i, :nnz],
-                  self.col[i, :nnz]] = data[i, :nnz]
```
Comment (hvy) [inline chainer/utils/sparse.py]: The above two lines were the cause of the issue for duplicate indices not taken into account.

File: chainer/utils/sparse.py [hunk 2]
```diff
@@ -1,5 +1,14 @@
+import numpy
+
 import chainer
 from chainer import backend
+from chainer import cuda
+
+
+def _add_at(add_at, x, row, col, data):
+    assert data.size > 0
+    last_nz = data.size - (data != 0)[::-1].argmax()
```
Comment (hvy) [inline chainer/utils/sparse.py]: This should correctly handle intermediate 0s in the data array, in contrast to the previous `nnz = xp.count_nonzero(data[i])`.

**Comment by hvy** [comment]
Discussed with @asi1024 that we should do some performance benchmarks before proceeding with the review. If this implementation is too slow, we might reconsider the expected behavior (e.g. keep the current behavior).

**Comment by asi1024** [comment]
@hvy A discussion about performance is left, but this PR looks good for a temporary fix. Can I merge this PR?

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**